src/mds/SessionMap.cc

```diff
@@ -43,6 +43,11 @@ class SessionMapIOContext : public MDSIOContextBase
 };
 };
 
+SessionMap::SessionMap(MDSRank *m)
+  : mds(m),
+    mds_session_metadata_threshold(g_conf().get_val<Option::size_t>("mds_session_metadata_threshold")) {
```

Comment:
@vshankar IIUC, this PR caused the failures we are seeing in the QA run.

Following is the traceback -
```
 1: /lib64/libpthread.so.0(+0x12b20) [0x7f2f1fe98b20]
 2: gsignal()
 3: abort()
 4: /lib64/libstdc++.so.6(+0x9009b) [0x7f2f1f29909b]
 5: /lib64/libstdc++.so.6(+0x9653c) [0x7f2f1f29f53c]
 6: /lib64/libstdc++.so.6(+0x96597) [0x7f2f1f29f597]
 7: /lib64/libstdc++.so.6(+0x967f8) [0x7f2f1f29f7f8]
 8: ceph-mds(+0x2145a3) [0x55667a0835a3]
 9: (Option::size_t const md_config_t::get_val<Option::size_t>(ConfigValues const&, std::basic_string_view<char, std::char_traits<char> >) const+0x5c) [0x55667a0f8bfc]
 10: (SessionMap::SessionMap(MDSRank*)+0x2c2) [0x55667a3419f2]
 11: (MDSRank::MDSRank(int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, ceph::fair_mutex&, std::shared_ptr<LogChannel>&, CommonSafeTimer<ceph::fair_mutex>&, Beacon&, std::unique_ptr<MDSMap, std::default_delete<MDSMap> >&, Messenger*, MonClient*, MgrClient*, Context*, Context*, boost::asio::io_context&)+0x214) [0x55667a05c954]
 12: (MDSRankDispatcher::MDSRankDispatcher(int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, ceph::fair_mutex&, std::shared_ptr<LogChannel>&, CommonSafeTimer<ceph::fair_mutex>&, Beacon&, std::unique_ptr<MDSMap, std::default_delete<MDSMap> >&, Messenger*, MonClient*, MgrClient*, Context*, Context*, boost::asio::io_context&)+0x101) [0x55667a05e3e1]
 13: (MDSDaemon::handle_mds_map(boost::intrusive_ptr<MMDSMap const> const&)+0x1608) [0x55667a02e368]
 14: (MDSDaemon::handle_core_message(boost::intrusive_ptr<Message const> const&)+0xcd) [0x55667a03146d]
 15: (MDSDaemon::ms_dispatch2(boost::intrusive_ptr<Message> const&)+0xc3) [0x55667a031fc3]
 16: (DispatchQueue::entry()+0x126a) [0x7f2f210f2e0a]
 17: (DispatchQueue::DispatchThread::entry()+0x11) [0x7f2f211a75b1]
 18: /lib64/libpthread.so.0(+0x814a) [0x7f2f1fe8e14a]
 19: clone()
```

Out of this following lines are important -
```
 9: (Option::size_t const md_config_t::get_val<Option::size_t>(ConfigValues const&, std::basic_string_view<char, std::char_traits<char> >) const+0x5c) [0x55667a0f8bfc]
 10: (SessionMap::SessionMap(MDSRank*)+0x2c2) [0x55667a3419f2]
 11: (MDSRank::MDSRank(int, std::__cxx11::basic_string<char, std::char_traits<char>, 
```

Out of all the PRs that are on the testing branch, these files have been modified by this one. Plus, the first lines draws attention to this line.

IMHO, the mistake that occurred here is that during rebase at `src/common/options/mds.yaml.in` was added as it is to resolve the conflict. Instead what needs to be done here is that this option (mds_session_metadata_threshold) needs to be added to `src/common/legacy_config_opts.h `. Since this option is not added to this file, it still is undefined which led this call to fail.

Reply:
Thanks @rishabh-d-dave. BTW, this PR wasn't ready for testing since additional it required additional commits to be picked. It got marked as needs-qa again after I removed the tag last week :/

Anyway, I'll redo the cherry-pick and push na update.